typos-ranking-rules.md

```diff
@@ -0,0 +1,59 @@
+# Typo and Ranking rules
+
+This is an explanation of the default rules used in MeiliDB.
+
+First of all we must explain some terms that are used in this reading.
+
+- A query string is the full list of all the words that the end user is searching for results.
+- A query word is one of the word that composes the query string.
+
+
+
+## Typo rules
+
+The typo rules are used before sorting the documents, they are used to aggregate them, to choose which documents contains words similar to the queried words.
+
+We use a prefix levenshtein algorithm to check if words match. The only difference with a Levenshtein algorithm is that it accepts every word that **starts with the query words** too. Therefore words are accepted if they starts with or have equal length.
+
+
+
+The Levenshtein distance between two words _M_ and _P_ is called the minimum cost of transforming _M_ into _P_ by performing the following elementary operations:
+
+- substitution of a character of _M_ by a character other than _P_. (e.g. **k**itten → **s**itten)
+- insertion in _M_ of a character of _P_. (e.g. sittin → sittin**g**)
+- deleting a character from _M_. (e.g. satu**r**day → satuday)
+
+
+
+There is some rules about what can be considered "similar". This rules are **by word** and not for the whole query string.
+
+- If the query word is between 1 and 4 long therefore **no** typo is allowed, only documents that contains words that start or are exactly equal to this query word are considered valid for this request.
+- If the query word is between 5 and 8 long, **one** typo is allowed, documents that contains words that match with one typo are used aggregated and use for the steps.
```

Comment:
If the query word is between 5 and 8 long, only one typo is allowed. Documents that contain words that match with one typo `are used for the next steps`. -> I am not sure what you mean at the end of the sentence